icu4c/source/i18n/measunit_impl.h

```diff
@@ -120,12 +119,14 @@ struct SingleUnitImpl : public UMemory {
     int32_t dimensionality = 1;
 };
 
+// For Windows: enable DLL export of a struct that uses this template.
+template class U_I18N_API MaybeStackVector<SingleUnitImpl>;
```

Comment:
Put it inside an ifdef, like this:

https://github.com/unicode-org/icu/blob/master/icu4c/source/i18n/collationiterator.h#L34

Reply:
Not yet sufficient:

```
icu4c\source\common\cmemory.h(759,0): Error C4251: 'icu_67::MemoryPool<icu_67::SingleUnitImpl,8>::fPool': class 'icu_67::MaybeStackArray<T *,8>' needs to have dll-interface to be used by clients of class 'icu_67::MemoryPool<icu_67::SingleUnitImpl,8>'
```

Reply:
Looks like I missed a "*".

Reply:
Or no, also a `MaybeStackArray<T*, stackCapacity>` for the private `fPool` member of MemoryPool. Adding that and the ifdef.

Reply:
OT (or tangential), I wonder if this solution found in number_decimfmtprops.h could be resolved by using a similar "template class" instantiation:

```
// Exported as U_I18N_API because it is a public member field of exported DecimalFormatProperties
// Using this wrapper is rather unfortunate, but is needed on Windows platforms in order to allow
// for DLL-exporting an fully specified template instantiation.
class U_I18N_API CurrencyPluralInfoWrapper {
```

Leaving it for now. (Is it worth filing a [low-priority] ticket for cruft / cleanup?)